cpp/src/io/parquet/experimental/hybrid_scan_chunking.cu

```diff
@@ -0,0 +1,338 @@
+
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "hybrid_scan_helpers.hpp"
+#include "hybrid_scan_impl.hpp"
+#include "io/comp/gpuinflate.hpp"
+#include "io/comp/io_uncomp.hpp"
+#include "io/comp/nvcomp_adapter.hpp"
+#include "io/parquet/compact_protocol_reader.hpp"
+#include "io/parquet/reader_impl_chunking.hpp"
+#include "io/utilities/time_utils.cuh"
+
+#include <cudf/detail/iterator.cuh>
+#include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/detail/utilities/integer_utils.hpp>
+#include <cudf/detail/utilities/vector_factories.hpp>
+#include <cudf/io/config_utils.hpp>
+#include <cudf/utilities/memory_resource.hpp>
+
+#include <rmm/exec_policy.hpp>
+
+#include <cuda/functional>
+#include <thrust/binary_search.h>
+#include <thrust/host_vector.h>
+#include <thrust/iterator/constant_iterator.h>
+#include <thrust/iterator/discard_iterator.h>
+#include <thrust/logical.h>
+#include <thrust/sort.h>
+#include <thrust/transform_scan.h>
+#include <thrust/unique.h>
+
+#include <numeric>
+
+namespace cudf::io::parquet::experimental::detail {
+
+namespace {
+
+namespace nvcomp = cudf::io::detail::nvcomp;
```

Comment:
Everything in this anonymous namespace is copy pasted as-is from `reader_impl_chunking.cu` so no need to review.